src/TabWidget.js

```diff
@@ -0,0 +1,386 @@
+          display: 'block',
+        });
+        const moreTab = query('.more-item', this.id)[0];
+        if (array.indexOf(this.tabList.children, tab) > -1) {
+          this.positionFocusState(tab);
+          domClass.toggle(this.currentTab, 'selected');
+          domClass.toggle(tab, 'selected');
+          this.currentTab = tab;
+          if (moreTab) {
+            domClass.remove(moreTab, 'selected');
+          }
+        } else {
+          if (moreTab) {
+            this.positionFocusState(moreTab);
+            domClass.add(moreTab, 'selected');
+            domClass.toggle(this.currentTab, 'selected');
+            domClass.toggle(tab, 'selected');
+            this.currentTab = tab;
+          }
+        }
+      }
+    }
+  },
+  /**
+   * Changes the tab state in the tab list and changes visibility of content.
+   * @param {Object} The event type and source.
+   */
+  toggleDropDown: function toggleDropDown(params) {
+    const tab = params.$source;
+    const moreTab = query('.more-item', this.id)[0];
+    const icon = query('.fa', moreTab)[0];
+
+    if (tab) {
+      if (domStyle.get(this.moreTabList, 'visibility') === 'hidden') {
+        domStyle.set(this.moreTabList, {
+          visibility: 'visible',
+        });
+        if (icon) {
+          domClass.replace(icon, 'fa fa-angle-down', 'fa fa-angle-right');
+        }
+
+        if (!this.moreTabList.style.left) {
+          const posTop = moreTab.offsetTop;
+          const posLeft = moreTab.offsetLeft;
+          const width = parseInt(moreTab.offsetWidth, 10);
+          const height = parseInt(moreTab.offsetHeight, 10);
+          const maxHeight = this.domNode.offsetHeight - this.domNode.offsetTop - posTop;
+
+          domStyle.set(this.moreTabList, {
+            left: posLeft - this.moreTabList.offsetWidth + width + 'px',
+            top: posTop + height + 'px',
+            maxHeight: maxHeight + 'px',
+          });
+        }
+      } else {
+        domStyle.set(this.moreTabList, {
+          visibility: 'hidden',
+        });
+        if (icon) {
+          domClass.replace(icon, 'fa fa-angle-right', 'fa fa-angle-down');
+        }
+      }
+    } else {
+      if (params.target !== moreTab) {
+        domStyle.set(this.moreTabList, {
+          visibility: 'hidden',
+        });
+        if (icon) {
+          domClass.replace(icon, 'fa fa-angle-right', 'fa fa-angle-down');
+        }
+      }
+    }
+  },
+  /**
+   * Reorganizes the tab when the screen orientation changes.
+   */
+  reorderTabs: function reorderTabs() {
+    const moreTab = query('.more-item', this.id)[0];
+    this.inOverflow = false;
+
+    if (this.moreTabList.children.length > 0) {
+      if (moreTab) {
+        this.tabList.children[this.tabList.children.length - 1].remove();
+      }
+      // Need to reference a different array when calling array.forEach since this.moreTabList.children is being modified, hence have arr be this.moreTabList.children
```

Comment:
Sound like we need to think of a way of getting the tabs in a formal and safe way vs just using the children collection.